roles/irods/tasks/icat.yml

```diff
@@ -1,29 +1,167 @@
+    creates: /etc/irods/{{ irods_ssl_dh_params_file }}
+  register: result
+  failed_when:
+    - result.rc != 0
+  become: true
+
+- name: Set ownership of {{ irods_ssl_dh_params_file }}
+  ansible.builtin.file:
+    path: '/etc/irods/{{ irods_ssl_dh_params_file }}'
+    owner: '{{ irods_service_account }}'
+    group: '{{ irods_service_account }}'
+    mode: '0644'
+  become: true
 
 - name: create_unattended_installation_json
-  become: True
   template:
     src: unattended_install.json.j2
     dest: /root/unattended_install.json
+    mode: '0600'
   register: create_unattended_installation_json
+  become: true
+
+# - name: Check if {{ irods_db_name }} database is available
+#  postgresql_ping:
+#    db: "{{ irods_db_name }}"
+#  register: check_database
+#  become_user: postgres
+#  become: true
+#
+- name: Create a new database "{{ irods_db_name }}" if missing
+  postgresql_db:
+    name: "{{ irods_db_name }}"
+    state: present
+#  when: check_database.is_available is false
+  register: db_created
+  become_user: postgres
+  become: true
 
-- name: run_irods_installation_script
-  become: True
-  shell:  python /var/lib/irods/scripts/setup_irods.py --json_configuration_file=/root/unattended_install.json
-  when: create_unattended_installation_json.changed
+- name: Creating irods sql user and granting privileges
+  postgresql_user:
+    name: "{{ irods_db_user }}"
+    password: "{{ irods_db_pwd }}"
+    db: "{{ irods_db_name }}"
+    priv: ALL
+    state: present
+  become_user: postgres
+  become: true
+
+- name: Collect list of tables from existing {{ irods_db_name }}
+  ansible.builtin.command: psql -d ICAT -c '\dt'
+  changed_when: false
+  register: database_tables
+  become_user: postgres
+  become: true
+
+- name: Install iRODS, if {{ irods_db_name }} is empty (no tables exist)
+  ansible.builtin.command: sudo python /var/lib/irods/scripts/setup_irods.py --json_configuration_file=/root/unattended_install.json
+  when: '"No relations found." in database_tables.stdout'
   run_once: true
   async: 120
-  poll: 1  
+  poll: 1
+  become_user: '{{ irods_service_account }}'
+  become: true
+
+- name: Edit /etc/irods/core.re to enforce the use of the certificates
+  become: true
+  lineinfile:
+    name: /etc/irods/core.re
+    search_string: 'acPreConnect(*OUT) { *OUT="CS_NEG_DONT_CARE"; }'
+    line: 'acPreConnect(*OUT) { *OUT="CS_NEG_REQUIRE"; }'
+
+- name: Edit /etc/irods/core.re to set number of threads to 4
+  become: true
+  lineinfile:
+    name: /etc/irods/core.re
+    search_string: 'acSetNumThreads {msiSetNumThreads("default","default","default"); }'
+    line: 'acSetNumThreads {msiSetNumThreads("default","4","default"); }'
+
+- name: Increase systcl limits to allow longer transfers
+  copy:
+    src: '{{ role_path }}/files/irods_fix.conf'
+    dest: /etc/sysctl.d/
+    mode: 0644
+  become: true
+
+- name: Fix the s3 authentication by creating dummy s3auth file
+  ansible.builtin.copy:
+    content: |
```

Comment:
When content is specified like this, we do not need the `roles/irods/files/s3auth` file anymore, right?

Reply:
Correct. Was doing it one way, then another. This is not needed anymore and will remove it :white_check_mark: